Application/Commands/RegisterSelectedCourseForApplicantCommand.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+using Domain.Aggreagtes.ApplicantAggregate;
+using Domain.Aggreagtes.CourseAggregate;
+using Domain.Enums;
+using Domain.Repositories;
+using MediatR;
+
+
+namespace Application.Commands
+{
+    public class RegisterSelectedCourseForApplicantCommand : IRequest<ApplicantEnrollmentResponse>
+    {
+        public Guid ApplicantId { get; init; }
+        public Guid CourseId { get; init; }
+        public Guid CourseAssessmentId { get; init; }
+        public CourseMode CourseMode { get; init; }
```

Comment:
what is CourseMode? why is it not using referenced by Id

Reply:
CourseMode object is among the constructor parameters of ApplicantEnrolllment , referencing through id will require that i retrieve a course having it in it courseMode list. I